compiler/rustc_query_system/src/dep_graph/graph.rs

```diff
@@ -1279,6 +1295,16 @@ pub enum TaskDepsRef<'a> {
     /// to ensure that the decoding process doesn't itself
     /// require the execution of any queries.
     Forbid,
+    /// Side effects from the previous run made available to
+    /// queries when they are reexecuted because their result was not
+    /// available in the cache. The query removes entries from the
+    /// side effect table. The table must be empty
```

Comment:
This comment is not completely consistent with the code. The last sentence is unfinished, did you mean to change it?